examples/stm32f7/src/bin/usart_dma.rs

```diff
@@ -0,0 +1,46 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+#[path = "../example_common.rs"]
+mod example_common;
+use core::fmt::Write;
+use embassy::executor::Executor;
+use embassy::util::Forever;
+use embassy_stm32::dma::NoDma;
+use embassy_stm32::usart::{Config, Uart};
+use embassy_traits::uart::Write as _Write;
+use example_common::*;
+
+use cortex_m_rt::entry;
+use heapless::String;
+
+#[embassy::task]
+async fn main_task() {
+    let p = embassy_stm32::init(Default::default());
+
+    let config = Config::default();
+    let mut usart = Uart::new(p.UART7, p.PA8, p.PA15, p.DMA1_CH1, NoDma, config);
+
+    for n in 0u32.. {
+        let mut s: String<128> = String::new();
+        core::write!(&mut s, "Hello DMA World {}!\r\n", n).unwrap();
+
+        usart.write(s.as_bytes()).await.ok();
```

Comment:
The error shouldn't be ignored like this, there is the `unwrap!()` macro from defmt, that will print the error and panic.

Using the macro won't however work on the `.unwrap()` on the line 27 as an implementation of defmt::Format would be required on the Error type. That is fine though.

Nice work!

Reply:
Thanks, will add that unwrap macro and the suggestion @huntc made and push the update.